.github/workflows/msbuild.yml

```diff
@@ -1,74 +1,79 @@
+---
 name: MSBuild
 
-on: [push]
+on:
+  pull_request:
+    branches: [master]
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [master]
+  workflow_dispatch:
 
-env:
-  # Path to the solution file relative to the root of the project.
-  SOLUTION_FILE_PATH: .
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
 
-  # Configuration type to build.
-  # You can convert this to a build matrix if you need coverage of multiple configuration types.
-  # https://docs.github.com/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
-  BUILD_CONFIGURATION: Release
+env:
+  SOLUTION_FILE_PATH: .  # Path to the solution file relative to the root of the project.
+  BUILD_CONFIGURATION: Release  # Configuration type to build.
 
 jobs:
   build:
     runs-on: windows-2022
-
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
       with:
         submodules: 'recursive'
-      
+
     - name: Add MSBuild to PATH
-      uses: microsoft/setup-msbuild@v1.0.2
+      uses: microsoft/setup-msbuild@v2
 
     - name: Restore VCPKG packages
-      working-directory: ${{env.GITHUB_WORKSPACE}}
-      shell: pwsh
-      run: Invoke-WebRequest -Uri "https://github.com/TheElixZammuto/moonlight-xbox/releases/download/1.10.0/vcpkg_installed.zip" -OutFile .\vcpkg_installed.zip
+      run: |
+        Invoke-WebRequest `
+          -Uri "https://github.com/TheElixZammuto/moonlight-xbox/releases/download/1.10.0/vcpkg_installed.zip" `
+          -OutFile .\vcpkg_installed.zip
+
     - name: Extract VCPKG packages
-      working-directory: ${{env.GITHUB_WORKSPACE}}
-      shell: pwsh
       run: Expand-Archive .\vcpkg_installed.zip -DestinationPath .\
-    - name: Extract VCPKG packages
-      working-directory: ${{env.GITHUB_WORKSPACE}}
-      shell: pwsh
+
+    - name: List VCPKG packages
       run: dir .\vcpkg_installed
-    - name: Complete install of folder
-      working-directory: ${{env.GITHUB_WORKSPACE}}
-      shell: pwsh
+
+    - name: Install VCPKG packages
       run: .\vcpkg\bootstrap-vcpkg.bat
-    #- name: Complete install of folder
-    #  working-directory: ${{env.GITHUB_WORKSPACE}}
-    #  shell: pwsh
-    #  run: .\vcpkg\vcpkg.exe install --triplet x64-uwp
+
     - name: Build third party tools
-      working-directory: ${{env.GITHUB_WORKSPACE}}
       run: .\generate-thirdparty-projects.bat
+
     - name: Restore NuGet
-      working-directory: ${{env.GITHUB_WORKSPACE}}
       run: nuget restore ${{env.SOLUTION_FILE_PATH}}
+
     - name: Load Certificate
+      env:
+        CERTIFICATE_FILE: ${{ secrets.CERTIFICATE_FILE }}
```

Comment:
I missed that there was a secret used in the workflow.

Can we load a dummy certificate when it's a PR?

Reply:
@TheElixZammuto any thoughts on this? My assumption is the certificate is only needed for production builds that will be sent to the MSFT Store?

Reply:
It's counterintuitive, but MS Store does not use the certificate inside secrets.CERTIFICATE_FILE, unlike App Store and Google Play. This is needed for install on Xboxes, and it's just a self signed certificate in pfx format

Reply:
Okay, so for PRs from forks, we could generate a certificate?

And for pushes or PRs from this repo, use the secret?

Reply:
I gave this an attempt, but the build is failing to validate the certificate.

```
        "D:\a\moonlight-xbox\moonlight-xbox\moonlight-xbox-dx.sln" (default target) (1) ->
       "D:\a\moonlight-xbox\moonlight-xbox\moonlight-xbox-dx.vcxproj" (default target) (2) ->
       (_ValidateSigningCertificate target) -> 
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VisualStudio\v17.0\AppxPackage\Microsoft.AppXPackage.Targets(894,5): error : Certificate could not be opened: cert.pfx. [D:\a\moonlight-xbox\moonlight-xbox\moonlight-xbox-dx.vcxproj]
       C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VisualStudio\v17.0\AppxPackage\Microsoft.AppXPackage.Targets(894,5): error : The specified network password is not correct. [D:\a\moonlight-xbox\moonlight-xbox\moonlight-xbox-dx.vcxproj]
       C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VisualStudio\v17.0\AppxPackage\Microsoft.AppXPackage.Targets(894,5): error :  [D:\a\moonlight-xbox\moonlight-xbox\moonlight-xbox-dx.vcxproj]

    3211 Warning(s)
    1 Error(s)
```

I'm not totally sure how the validation occurs, but it seems that it's trying to make a request to the address.

Most of what I tried is from here: https://stackoverflow.com/questions/11403333